### PR TITLE
Add swift module

### PR DESCRIPTION
**Pull request by paladox**

### CODE REVIEW

**Comment by paladox** [comment]
DO NOT MERGE, it's not ready and basically an old version.

File: modules/swift/manifests/backend.pp [hunk 1]
```diff
@@ -0,0 +1,111 @@
+# == Class: swift::backend
+
+class swift::backend (
+    $active_server = hiera('swift_active_server', true),
+) {
+
+    require_package(['swift-account', 'swift-container', 'swift-object'])
```
Comment (Universal-Omega) [inline modules/swift/manifests/backend.pp:7]:
```suggestion
    ensure_packages(['swift-account', 'swift-container', 'swift-object'])
```

**Review by Universal-Omega — CHANGES_REQUESTED** [review]

File: manifests/site.pp [hunk 2]
```diff
@@ -108,6 +108,21 @@
...
+    include base
+    include role::swift
+}
+
+node 'swiftac111.miraheze.org' {
+    include base
+    include role::swift
+}
+
+node /^swiftobject[12345]\.miraheze\.org$/ {
```
Comment (Universal-Omega) [inline manifests/site.pp:121]:
```suggestion
node /^swiftobject11[12345]\.miraheze\.org$/ {
```

**Review by Universal-Omega — CHANGES_REQUESTED** [review]

File: modules/rsync/manifests/quickdatacopy.pp [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+#
+# [*auto_interval*] If $auto_sync is true, the interval to sync at. Defaults to every 10 minutes. See
+#                   systemd::timer::job's $interval parameter and Systemd::Timer::Schedule for more details.
+define rsync::quickdatacopy(
+  Stdlib::Fqdn $source_host,
+  Stdlib::Fqdn $dest_host,
+  Stdlib::Unixpath $module_path,
+  Optional[Stdlib::Unixpath] $file_path = undef,
+  Boolean $auto_sync = true,
+  Wmflib::Ensure $ensure = present,
```
Comment (Universal-Omega) [inline modules/rsync/manifests/quickdatacopy.pp:47]:
```suggestion
  VMlib::Ensure $ensure = present,
```

**Pull request merged**